src/Features/CSharp/Portable/Snippets/CSharpForLoopSnippetProvider.cs

```diff
@@ -0,0 +1,135 @@
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var iteratorName = NameGenerator.GenerateUniqueName(
+                new List<string> { "i", "j", "k", "a", "b", "c" },
+                n => semanticModel.LookupSymbols(position, name: n).IsEmpty);
+            var indexVariable = generator.Identifier(iteratorName);
+
+            // Creating the variable declaration based on if the user has
+            // 'var for built in types' set in their editorconfig.
+            var variableDeclarationSyntax =
+                 SyntaxFactory.VariableDeclaration(compilation.GetSpecialType(SpecialType.System_Int32).GenerateTypeSyntax(allowVar: false),
+                    SyntaxFactory.SingletonSeparatedList(
+                        SyntaxFactory.VariableDeclarator(
+                            indexVariable,
+                            argumentList: null,
+                            SyntaxFactory.EqualsValueClause((ExpressionSyntax)generator.LiteralExpression(0)))));
+
+            var forLoopSyntax = SyntaxFactory.ForStatement(
+                variableDeclarationSyntax,
+                SyntaxFactory.SeparatedList<ExpressionSyntax>(),
+                (ExpressionSyntax)generator.LessThanExpression(
+                    generator.IdentifierName(indexVariable),
+                    // Using a temporary identifier name for now, could later be changed
+                    // to look for an iterable item in the scope of the insertion.
+                    generator.IdentifierName("length")),
+                SyntaxFactory.SingletonSeparatedList<ExpressionSyntax>(
+                    SyntaxFactory.PostfixUnaryExpression(
+                        SyntaxKind.PostIncrementExpression, SyntaxFactory.IdentifierName(indexVariable))),
+                SyntaxFactory.Block());
+
+            return forLoopSyntax;
+        }
+
+        /// <summary>
+        /// Goes through each piece of the for statement and extracts the identifiers
+        /// as well as their locations to create SnippetPlaceholder's of each.
+        /// </summary>
+        protected override ImmutableArray<SnippetPlaceholder> GetPlaceHolderLocationsList(SyntaxNode node, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<SnippetPlaceholder>.GetInstance(out var arrayBuilder);
+            var placeHolderBuilder = new MultiDictionary<string, int>();
+            GetPartsOfForStatement(node, out var declaration, out var condition, out var incrementor, out var _1);
+
+            // Retrieves the placeholder present in the variable declaration as well as its location.
+            // The declaration is constructed so it can't be null.
+            var variableDeclarator = ((VariableDeclarationSyntax)declaration!).Variables.FirstOrDefault();
```

Comment:
small nit: if it can't be null, then just call `First()` or `Single()`

Reply:
👍